test/com/m3/octoparts/util/OctoMetricsImplSpec.scala

```diff
@@ -0,0 +1,44 @@
+package com.m3.octoparts.util
+
+import com.codahale.metrics.JvmAttributeGaugeSet
+import com.codahale.metrics.jvm.{ GarbageCollectorMetricSet, MemoryUsageGaugeSet, ThreadStatesGaugeSet }
+import com.m3.octoparts.support.PlayAppSupport
+import org.scalatest.{ BeforeAndAfterEach, Matchers, FunSpec }
+import scala.collection.JavaConverters._
+
+class OctoMetricsImplSpec extends FunSpec with Matchers with PlayAppSupport with BeforeAndAfterEach {
+
+  override protected def afterEach(): Unit = {
+    subject.onStop()
+  }
+
+  lazy val subject = new OctoMetricsImpl(appComponents.applicationLifecycle, appComponents.configuration)
+
+  describe("#onStart") {
+
+    it("should not throw") {
+      subject.onStart()
+    }
+
+    it("should work with consecutive calls") {
+      subject.onStart()
+      subject.onStart()
+    }
+
+    it("should work if metrics were already registered") {
```

Comment:
More pedantry and my own confirmation, this is checking if the _same_ metrics were already registered on the same default registry, yes?

A slight suggestion would be to create the subject with `jvmMetricsEnabled` true, and just call `setupJvmMetrics` before the `onStart` call. Unfortunately the specific attributes that get enabled aren't factored into a list in Kenshoo so the `find`s will have to remain =/